hpx/runtime/agas/big_boot_barrier.hpp

```diff
@@ -125,7 +125,8 @@ struct HPX_EXPORT big_boot_barrier : boost::noncopyable
 
     void add_thunk(util::unique_function_nonser<void()>* f)
     {
-        thunks.push(f);
+        while(!thunks.push(f))
+            ; // Wait until succesfully pushed ...
```

Comment:
> push is guaranteed to succeed eventually, iiuc. Should we really time out and throw here?

Not sure. I usually try to avoid things like that. Smells like a live-lock...